syntax = "proto3";


message LoanPoint {
    string rate = 1;
    string amount = 2;
    int32 range_min = 3;
    int32 range_max = 4;
}


message Orders {
    repeated LoanPoint offers = 1;
    repeated LoanPoint demands = 2;
}


message Currency {
    string name = 1;
}


service Gateway {
    rpc loan_orders(Currency) returns (Orders) {}
}
